internal/manager/pvcleanup/controller.go

```diff
@@ -0,0 +1,227 @@
+		if apierrors.IsNotFound(err) {
+			// PV was deleted, nothing to do
+			return ctrl.Result{}, nil
+		}
+		logger.Error(err, "failed to get PersistentVolume")
+		return ctrl.Result{}, err
+	}
+
+	// Only process PVs from our CSI driver
+	if pv.Spec.CSI == nil || pv.Spec.CSI.Driver != lvm.DriverName {
+		logger.V(4).Info("PV is not from our CSI driver, skipping", "pv", pv.Name, "driver", pv.Spec.CSI)
+		return ctrl.Result{}, nil
+	}
+
+	// Only process PVs that are Released with Delete reclaim policy
+	if pv.Status.Phase != corev1.VolumeReleased {
+		// PV is not released, nothing to do
+		return ctrl.Result{}, nil
+	}
+
+	if pv.Spec.PersistentVolumeReclaimPolicy != corev1.PersistentVolumeReclaimDelete {
+		// PV doesn't have Delete reclaim policy, nothing to do
+		logger.V(4).Info("PV doesn't have Delete reclaim policy, skipping", "pv", pv.Name, "reclaimPolicy", pv.Spec.PersistentVolumeReclaimPolicy)
+		return ctrl.Result{}, nil
+	}
+
+	hasPVProtection := hasFinalizer(&pv, PVProtectionFinalizer)
+	hasExternalProvisioner := hasFinalizer(&pv, ExternalProvisionerFinalizer)
+
+	if !hasPVProtection && !hasExternalProvisioner {
+		// PV doesn't have any finalizers we care about, nothing to do
+		return ctrl.Result{}, nil
+	}
+
+	logger.Info("PV is released with delete reclaim policy", "pv", pv.Name, "driver", pv.Spec.CSI.Driver)
+
+	// Extract hostnames from PV's node affinity
+	hostnames := extractHostnamesFromPV(&pv)
+	if len(hostnames) == 0 {
+		logger.Info("PV has no hostname topology constraints, skipping finalizer removal", "pv", pv.Name)
+		return ctrl.Result{}, nil
+	}
+
+	logger.Info("PV has hostname topology", "pv", pv.Name, "hostnames", hostnames)
+
+	// Check if any of the hostnames are available as nodes in the cluster
+	hasAvailableNode := false
+	for _, hostname := range hostnames {
+		var node corev1.Node
+		if err := r.Get(ctx, client.ObjectKey{Name: hostname}, &node); err == nil {
+			// Node exists, check if it's ready
+			if isNodeReady(&node) {
+				hasAvailableNode = true
+				logger.Info("Found available node for PV", "pv", pv.Name, "node", hostname)
+				break
+			}
+		}
+	}
+
+	if hasAvailableNode {
+		logger.Info("PV has at least one available node, keeping finalizer", "pv", pv.Name)
+		return ctrl.Result{}, nil
+	}
+
+	logger.Info("None of the PV's hostname nodes are available, removing finalizers", "pv", pv.Name, "hostnames", hostnames)
+
+	// Remove the finalizers
+	patch := client.MergeFrom(pv.DeepCopy())
+	finalizersToRemove := []string{}
+	if hasPVProtection {
+		finalizersToRemove = append(finalizersToRemove, PVProtectionFinalizer)
+	}
+	if hasExternalProvisioner {
+		finalizersToRemove = append(finalizersToRemove, ExternalProvisionerFinalizer)
+	}
+
+	for _, finalizer := range finalizersToRemove {
+		pv.Finalizers = removeFinalizer(pv.Finalizers, finalizer)
+	}
+
+	if err := r.Patch(ctx, &pv, patch); err != nil {
+		logger.Error(err, "failed to remove finalizers from PV", "pv", pv.Name)
```

Comment:
I think that controller-runtime will print out these errors, so it wouldn't be necessary to log them